apps/dashboard/src/pages/api-keys.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState } from 'react';
+import { RiKey2Line, RiEyeLine, RiEyeOffLine, RiLockStarFill, RiExternalLinkLine } from 'react-icons/ri';
+import { useEnvironment } from '@/context/environment/hooks';
+import { CopyButton } from '@/components/primitives/copy-button';
+import { Card, CardContent } from '@/components/primitives/card';
+import { Button } from '@/components/primitives/button';
+import { InputField } from '@/components/primitives/input';
+import { Form } from '@/components/primitives/form/form';
+import { useForm } from 'react-hook-form';
+import { DashboardLayout } from '../components/dashboard-layout';
+import { PageMeta } from '@/components/page-meta';
+import { useApiKeysQuery } from '../hooks/use-api-keys-query';
+
+interface ApiKeysFormData {
+  apiKey: string;
+  environmentId: string;
+  identifier: string;
+}
+
+export function ApiKeysPage() {
+  const apiKeysQuery = useApiKeysQuery();
+  const { currentEnvironment } = useEnvironment();
+  const [showApiKey, setShowApiKey] = useState(false);
+  const apiKeys = apiKeysQuery.data?.data;
+
+  const form = useForm<ApiKeysFormData>({
+    values: {
+      apiKey: apiKeys?.[0]?.key ?? '',
+      environmentId: currentEnvironment?._id ?? '',
+      identifier: currentEnvironment?.identifier ?? '',
+    },
+  });
+
+  if (!currentEnvironment) {
+    return null;
+  }
+
+  const toggleApiKeyVisibility = () => {
+    setShowApiKey(!showApiKey);
+  };
+
+  const maskApiKey = (key: string) => {
+    return `${'•'.repeat(28)}${key.slice(-4)}`;
+  };
+
+  return (
+    <>
+      <PageMeta title="Environment Keys" />
+      <DashboardLayout headerStartItems={<h1 className="text-foreground-950">API Keys</h1>}>
+        <div className="flex flex-col gap-6 p-2">
+          <div className="grid grid-cols-1 gap-6 pt-10 lg:grid-cols-[1fr,500px]">
+            <Form {...form}>
+              <Card className="shadow-none">
+                <CardContent className="p-6">
+                  <div className="space-y-4">
+                    <div className="space-y-2">
+                      <label className="text-foreground-600 text-sm font-medium">API Key</label>
+                      <div className="flex items-center gap-2">
+                        <InputField className="flex-1">
+                          <div className="text-foreground-600">
+                            <RiKey2Line className="size-4" />
+                          </div>
+                          <input
+                            readOnly
+                            value={showApiKey ? form.getValues('apiKey') : maskApiKey(form.getValues('apiKey'))}
```

Comment:
There is a small UI glitch on the masked value. The last 6 dots are higher than the rest.

<img width="412" alt="Screenshot 2024-12-04 at 15 41 22" src="https://github.com/user-attachments/assets/da4a1b82-898d-4bad-866a-c48abb757e0d">